src/store/messagesStore.js

```diff
@@ -521,10 +522,12 @@ const actions = {
 			const parentInStore = context.getters.message(message.token, message.parent.id)
 			if (Object.keys(parentInStore).length !== 0) {
 				context.commit('addMessage', message.parent)
-				context.dispatch('resetReactions', {
-					token: message.token,
-					messageId: message.parent.id,
-				})
+				const reactionsStore = useReactionsStore()
+				if (message.systemMessage.startsWith('reaction')) {
+					reactionsStore.fetchReactions(message.token, message.parent.id)
```

Comment:
If we get 50 reactions on a single message in a row, that means we fetch details 50 times.
- In history conversation - messages come at once, and we fetch them instantly as many times, as there were reactions added/deleted

![image](https://github.com/nextcloud/spreed/assets/93392545/81608e15-e511-4a3b-8d7a-d736b8144407)

=> Solution here could be comparing new and old reactions object (`JSON.stringify`)

- In currently ongoing conversation - each participant online will receive all details each time somebody reacts. That means - <participants online> * <reactions sent> requests to server, while not everyone is going to need that info (for example, I see amount of reactions, but am not interested, who sent it)

=> Not sure about the right way here, but maybe I'd stick to the old flow with some changes - keep simplified reactions and load them on hover only (but hover on emojis buttons, not the message body)

Reply:
> Solution here could be comparing new and old reactions object (JSON.stringify)

This would be a solution to minimize store update but not fetch requests.

> Not sure about the right way here, but maybe I'd stick to the old flow with some changes - keep simplified reactions and load them on hover only (but hover on emojis buttons, not the message body)

This means make `Reactions` dependent on `Message` again through reactions prop. I thought we are trying to avoid that. Also, this brings back the issue of count is showing a different number then the number of participants in the details list.

Reply:
> This would be a solution to minimize store update but not fetch requests.

Not if we compare `message.reactions` and `parentInStore.reactions`

> Also, this brings back the issue of count is showing a different number then the number of participants in the details list.

Issue was that within 30 interval between messages polling, messageStore object has old value (`reactions: { 👍: 1, 😃: 1 })`) and by hover we fetching actual details (`reactions: { 👍: [admin], 😃: [bob, alice] })`). Wouldn't it be easier to compare them on hover, and patch message object with `😃: [bob, alice].length`?

Reply:
Is there a possibility to get reaction message sent as reaction_revoked message and not with polling messages process ? This would solve it from the root.

Reply:
I'd raise an issue for API here, to include either reaction itself, or ID of message which adds this reaction and is replaced with `reaction_deleted`